Branch-SDK/src/io/branch/referral/ApkParser.java

```diff
@@ -98,41 +105,56 @@ public String decompressXML(byte[] xml) {
         //   3rd word: Flags?
         //   4th word: str ind of attr value again, or ResourceId of value
 
-        // Step through the XML tree element tags and attributes
-        String attrValue;
-        String attrName;
-        int off = xmlTagOff;
-        while (off < xml.length) {
-            int tag0 = LEW(xml, off);
-            if (tag0 == startTag) { // XML START TAG
-                int numbAttrs = LEW(xml, off+7*4);  // Number of Attributes to follow
-                off += 9*4;  // Skip over 6+3 words of startTag data
-                // Look for the Attributes
-                for (int ii=0; ii<numbAttrs; ii++) {
-                    int attrNameSi = LEW(xml, off+1*4);  // AttrName String Index
-                    int attrValueSi = LEW(xml, off+2*4); // AttrValue Str Ind, or FFFFFFFF
-                    int attrResId = LEW(xml, off+4*4);  // AttrValue ResourceId or dup AttrValue StrInd
-                    off += 5*4;  // Skip over the 5 words of an attribute
-
-                    attrName = compXmlString(xml, sitOff, stOff, attrNameSi);
-                    if (attrName.equals("scheme")) {
-                        attrValue = attrValueSi!=-1 ? compXmlString(xml, sitOff, stOff, attrValueSi) : "resourceID 0x"+Integer.toHexString(attrResId);
-                        if (validURI(attrValue))
-                            return attrValue;
+        try {
+            // Step through the XML tree element tags and attributes
+            String attrValue;
+            String attrName;
+            int off = xmlTagOff;
+            while (off < xml.length) {
+                int tag0 = LEW(xml, off);
+                if (tag0 == startTag) { // XML START TAG
+                    int numbAttrs = LEW(xml, off+7*4);  // Number of Attributes to follow
+                    off += 9*4;  // Skip over 6+3 words of startTag data
+                    // Look for the Attributes
+                    for (int ii=0; ii<numbAttrs; ii++) {
+                        int attrNameSi = LEW(xml, off+1*4);  // AttrName String Index
+                        int attrValueSi = LEW(xml, off+2*4); // AttrValue Str Ind, or FFFFFFFF
+                        int attrResId = LEW(xml, off+4*4);  // AttrValue ResourceId or dup AttrValue StrInd
+                        off += 5*4;  // Skip over the 5 words of an attribute
+
+                        attrName = compXmlString(xml, sitOff, stOff, attrNameSi);
+                        if (attrName.equals("scheme")) {
+                            attrValue = attrValueSi != -1 ? compXmlString(xml, sitOff, stOff, attrValueSi) : "resourceID 0x" + Integer.toHexString(attrResId);
+                            if (validURI(attrValue)) {
+                                intentFilters.put("scheme", attrValue);
```

Comment:
Are we reading only one scheme her? An intent filter can have multiple schemes.  
Also hosts are associated with a scheme. This code doesnot seem to make a relation between a scheme and hosts